File: dart/greeter/lib/greeter.dart
import 'dart:io';

class InputHandler {
  static String? getUserName() {
    stdout.write("Enter your name: ");
    String? userInput = stdin.readLineSync();

    String? name = userInput?.trim();

    if (name == null || name.isEmpty) {
      name = "Stranger";
    }

    return name;
  }
}

class Greeter {
  String username;

  Greeter(this.username);

  void greet() {
    print("Hello, $username!");
    print("Greetings from Dart!");
  }
}

void main() {
  String? name = InputHandler.getUserName();
  Greeter greeter = Greeter(name!);

  greeter.greet();
}

File: scala/greeter/src/main/scala/Greeter.scala
package greeter

import scala.io.StdIn.readLine

object InputHandler:
  def getUserName(): String =
    print("Enter your name: ")
    val userInput = readLine()

    val name = userInput.trim

    if name.isEmpty then "Stranger" else name

class Greet(name: String):
  def greet(): Unit =
    println(s"Hello, $name!")
    println("Greetings from Scala!")

@main def Greeter(): Unit =
  val name = InputHandler.getUserName()
  val greeter = new Greet(name)

  greeter.greet()
